pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -207,7 +207,16 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 	}
 
 	// Update our Status based on the state of our underlying Route.
-	service.Status.PropagateRouteStatus(route.Status)
+	var validateFunc func() bool
+	switch {
+	case service.Spec.RunLatest != nil:
+		// In case of RunLatest, verify that traffic has already migrated
+		// to the latest revision.
+		validateFunc = func() bool {
+			return len(route.Status.Traffic) > 0 && route.Status.Traffic[0].RevisionName == config.Status.LatestReadyRevisionName
```

Comment:
Not a huge fan of passing in a function here, which kind of requires the caller to know the guts of what its calling.

I wonder if we could accomplish something that reads more smoothly by checking this stuff first and calling new `MarkFoo` methods, and otherwise falling back on `PropagateRouteStatus`?

Reply:
Discussed in person. Opted for postprocessing, for clarity, though, imo, the difference is negligible and there is more code. :-)